node/pkg/watchers/evm/custom_consistency_level.go

```diff
@@ -0,0 +1,340 @@
+	data, err := w.cclReadContract(ctx, emitterAddr)
+	if err != nil {
+		return &NothingSpecial{}, err
+	}
+
+	if data == cclEmptyData {
+		return &NothingSpecial{}, nil
+	}
+
+	request, err := cclParseConfig(data)
+	if err != nil {
+		return &NothingSpecial{}, fmt.Errorf("failed to parse contract data: %w", err)
+	}
+
+	return request, err
+}
+
+// cclReadContract calls into the contract to read the configuration for a given emitter.
+func (w *Watcher) cclReadContract(ctx context.Context, emitterAddr ethCommon.Address) ([32]byte, error) {
+	// Before we read the config from the contract, see if we already have it in the cache.
+	data, found := w.cclCacheLookUp(emitterAddr)
+	if found {
+		return data, nil
+	}
+
+	data, err := CCLReadContract(ctx, w.ethConn.Client(), w.cclAddr, emitterAddr)
+	if err != nil {
+		return cclEmptyData, err
+	}
+
+	w.cclCacheUpdate(emitterAddr, data)
+	w.cclLogger.Debug("read contract", zap.Stringer("emitterAddr", emitterAddr))
+	return data, nil
+}
+
+// CCLReadContract calls into the contract to read the configuration for a given emitter.
+// This is a free function so it can be called by the config verification tool.
+func CCLReadContract(ctx context.Context, ethClient *ethclient.Client, cclAddr ethCommon.Address, emitterAddr ethCommon.Address) ([32]byte, error) {
+	timeout, cancel := context.WithTimeout(ctx, 15*time.Second)
+	defer cancel()
+
+	caller, err := cclAbi.NewCclCaller(cclAddr, ethClient)
+	if err != nil {
+		return cclEmptyData, fmt.Errorf("failed to create abi caller: %w", err)
+	}
+
+	data, err := caller.GetConfiguration(&ethBind.CallOpts{Context: timeout}, emitterAddr)
+	if err != nil {
+		return cclEmptyData, err
+	}
+
+	return data, nil
+}
+
+// cclParseConfig parses the configuration data returned by the contract into a request.
+func cclParseConfig(data [32]byte) (CCLRequest, error) {
+	reader := bytes.NewReader(data[:])
+
+	t := CCLRequestType(0)
+	if err := binary.Read(reader, binary.BigEndian, &t); err != nil {
+		return nil, fmt.Errorf("failed to read data type: %w", err)
+	}
+
+	if t == 0x01 {
+		return cclParseAdditionalBlocksConfig(reader)
+	}
+
+	if t == 0x00 {
+		return &NothingSpecial{}, nil
+	}
+
+	return nil, fmt.Errorf("unexpected data type: %d", t)
+}
+
+// cclParseAdditionalBlocksConfig parses the configuration for an additional blocks request.
+// Note that the configuration type (the first byte) has already been read and verified.
+func cclParseAdditionalBlocksConfig(reader *bytes.Reader) (CCLRequest, error) {
+	consistencyLevel := uint8(0)
+	if err := binary.Read(reader, binary.BigEndian, &consistencyLevel); err != nil {
+		return nil, fmt.Errorf("failed to read consistency level: %w", err)
+	}
+
+	blocks := uint16(0)
+	if err := binary.Read(reader, binary.BigEndian, &blocks); err != nil {
+		return nil, fmt.Errorf("failed to read num blocks: %w", err)
+	}
+
+	// The config data is 32 bytes and the AdditionalBlocks request uses the first four, so we should have 28 left.
```

Comment:
my impression of the earlier comment was to check that these 28 bytes were zeroes as well, but I could be mistaken (also I personally don't think that's critical to check since the spec doesn't specify what needs to be in the unused bytes)